### PR TITLE
stitch: Omit ssh keys from machine hash calculation

### PR DESCRIPTION
**Pull request by mmussomele**

### CODE REVIEW

File: stitch/bindings.js [hunk 1]
```diff
@@ -23,12 +26,19 @@ function Deployment(deploymentOpts) {
     this.invariants = [];
 }
 
+function omit_sshkey(key, value) {
```
Comment (Baisang) [inline stitch/bindings.js]: nit as we don't really follow a specific JS style yet, but if we're going to camelcase should be `omitSshKey` or something (man that looks ugly though)

File: stitch/bindings.js.go [hunk 2]
```diff
@@ -27,12 +30,19 @@ function Deployment(deploymentOpts) {
     this.invariants = [];
 }
 
+function omitSSHKey(key, value) {
+    if (key == sshKeysField) {
```
Comment (kklin) [inline stitch/bindings.js.go]: I don't think `sshKeysField` needs to be a variable, it'd be slightly easier to read if you just used a string constant here IMO

File: quiltctl/command/run_test.go [hunk 3]
```diff
@@ -38,7 +38,7 @@ func TestRunSpec(t *testing.T) {
 	stitch.DefaultImportGetter.Path = "/quilt_path"
```
Comment (kklin) [inline quiltctl/command/run_test.go]: Can you add why we made this change to the commit message?

File: stitch/bindings.js.go [hunk 4]
```diff
@@ -27,12 +30,19 @@ function Deployment(deploymentOpts) {
...
+        return undefined;
+    }
+    return value;
+}
+
 // key creates a string key for objects that container a _refID, namely Containers
```
Comment (kklin) [inline stitch/bindings.js.go]: While you're in here, can you fix this typo? ("objects that _container_ a _refID" -> "objects that _have_ a _refID")

File: stitch/stitch_test.go [hunk 5]
```diff
@@ -38,17 +38,40 @@ func TestMachine(t *testing.T) {
 				SSHKeys:  []string{"key1", "key2"},
 			}})
 
+	checkMachines(t, `deployment.deploy([new Machine({
```
Comment (kklin) [inline stitch/stitch_test.go]: I suspect we'll forget what this is testing, a comment would be helpful.

**Comment by mmussomele** [comment]
@kklin updated with your comments.

**Pull request merged**